File: processor/src/test/resources/approvals/beanvalidation/QuantityValueExtractor.java
package test;

import jakarta.validation.valueextraction.ExtractedValue;
import jakarta.validation.valueextraction.ValueExtractor;
import scenarios.java.Quantity;

import javax.annotation.processing.Generated;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.beanvalidation.BeanValidationGenerator")
public class QuantityValueExtractor implements ValueExtractor<@ExtractedValue(type = Integer.class) Quantity> {
  @Override
  public void extractValues(Quantity originalValue, ValueExtractor.ValueReceiver receiver) {
    if (originalValue == null) {
      receiver.value(null, null);
      return;
    }
    receiver.value(null, originalValue.value());
  }
}

File: ksp/src/test/resources/approvals/beanvalidation/QuantityValueExtractor.kt
package test.domain

import jakarta.`annotation`.Generated
import jakarta.validation.valueextraction.ValueExtractor
import jakarta.validation.valueextraction.UnwrapByDefault
import scenarios.kotlin.Quantity

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.beanvalidation.BeanValidationGenerator")
@UnwrapByDefault
public class QuantityValueExtractor : QuantityValueExtractorBase() {
  override fun extractValues(originalValue: Quantity?, `receiver`: ValueExtractor.ValueReceiver) {
    if (originalValue == null) {
      receiver.value(null, null)
      return
    }
    receiver.value(null, originalValue.value)
  }
}
